app/models/concerns/subform_utilities.rb

```diff
@@ -6,8 +6,13 @@ module SubformUtilities
   end
 
   module ClassMethods
-    def distinct_by_column(name)
-      select("distinct on (#{self.name.underscore.pluralize}.#{name}) #{self.name.underscore.pluralize}.*")
+    def distinct_by_column(column)
+      raise StandardError.new("Column name invalid!") unless column_names.include?(column)
+      tbl_name = name.tableize
+      sql_select = send(
+        :sanitize_sql, ["DISTINCT ON (#{tbl_name}.#{column}) #{tbl_name}.*"]
```

Comment:
Not sure what `sanitize_sql` does ?

Reply:
It creates a sql query. For safety it's done @abulsayyad123 .

Reply:
https://apidock.com/rails/ActiveRecord/Sanitization/ClassMethods/sanitize_sql_for_conditions

`sanitize_sql` is an alias for `sanitize_sql_for_conditions`